taraxa/state/state_transition/state_transition.go

```diff
@@ -93,18 +104,29 @@ func (self *StateTransition) evm_state_checkpoint() {
 func (self *StateTransition) BeginBlock(blk_info *vm.BlockInfo) {
 	self.begin_block()
 	blk_n := self.pending_blk_state.GetNumber()
-	rules_changed := self.evm.SetBlock(&vm.Block{blk_n, *blk_info}, self.chain_cfg.ETHChainConfig.Rules(blk_n))
+	rules_changed := self.evm.SetBlock(&vm.Block{blk_n, *blk_info}, self.chain_config.ETHChainConfig.Rules(blk_n))
 	if self.dpos_contract != nil && rules_changed {
 		self.dpos_contract.Register(self.evm.RegisterPrecompiledContract)
 	}
-	if self.chain_cfg.ETHChainConfig.IsDAOFork(blk_n) {
+	if self.chain_config.ETHChainConfig.IsDAOFork(blk_n) {
```

Comment:
Cannot we simply use this whole structure that is already implemented for hardforks inside evm ? You could add this new hardfork to the struct that already exists:

```
type ChainConfig struct {
	HomesteadBlock types.BlockNum `json:"homesteadBlock,omitempty"`
	DAOForkBlock   types.BlockNum `json:"daoForkBlock,omitempty"`   // TheDAO hard-fork
	// EIP150 implements the Gas price changes (https://github.com/ethereum/EIPs/issues/150)
	EIP150Block         types.BlockNum `json:"eip150Block,omitempty"`         // EIP150 HF block (nil = no fork)
	EIP158Block         types.BlockNum `json:"eip158Block,omitempty"`         // EIP158 HF block
	...
}
```

no need to have hardfork.go file

Reply:
I am not sure that it is a good idea. I did it to separate our hardforks from Ethereum hardforks, that seems like a better option for me

Reply:
yes, we should not combine this

Reply:
what a ethereum hardfork in our network ? We dont have any because they are switched off and this repository is so heavily updated without thinking of keeping the original code separated that this is not argument anymore. So what is theadvantage of keeping eth hardfork separated from ours ? @MatusKysel @kstdl

Reply:
eth hardfork defines how our EVM should behave. We have all of them (that are part of our EVM), enabled on block 0